pkg/rbd/controllerserver.go

```diff
@@ -265,93 +223,75 @@ func (cs *ControllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 	}
 	// For now the image get unconditionally deleted, but here retention policy can be checked
 	volumeID := req.GetVolumeId()
+	if volumeID == "" {
+		return nil, status.Error(codes.InvalidArgument, "Empty volume ID in request")
+	}
 	volumeIDMutex.LockKey(volumeID)
-
 	defer func() {
 		if err := volumeIDMutex.UnlockKey(volumeID); err != nil {
 			klog.Warningf("failed to unlock mutex volume:%s %v", volumeID, err)
 		}
 	}()
 
 	rbdVol := &rbdVolume{}
-	if err := cs.MetadataStore.Get(volumeID, rbdVol); err != nil {
-		if err, ok := err.(*util.CacheEntryNotFound); ok {
-			klog.V(3).Infof("metadata for volume %s not found, assuming the volume to be already deleted (%v)", volumeID, err)
+	if err := genVolFromVolID(rbdVol, volumeID, req.GetSecrets()); err != nil {
+		// If image key is missing, there can be no unreserve as request name is unknown, and also
```

Comment:
some how failing to understand `no unreserve..etc` , may be reword it ?

Reply:
Updated comments in the area appropriately.